ffi/CppTest.hs

```diff
@@ -11,19 +11,37 @@ run in ghci using:
 {-# LANGUAGE TemplateHaskell #-}
 {-# LANGUAGE QuasiQuotes #-}
 {-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE OverloadedStrings #-}
 
 module Main where
 
 import qualified Language.C.Inline.Cpp as C
 import qualified Language.C.Inline.Cpp.Exceptions as C
+import qualified Language.C.Inline.Context as C
+import qualified Language.C.Types as C
+import qualified Data.Map as Map
 
-C.context C.cppCtx
+import Foreign.C.String
+import Foreign.C.Types
+import Foreign
+
+data Tensor
+
+C.context $ C.cppCtx <> mempty {
+    C.ctxTypesTable = Map.fromList [
+      (C.TypeName "PtrTensor", [t|Ptr Tensor|])
+    ]
+}
 
 C.include "<iostream>"
 C.include "<torch/torch.h>"
 C.include "<torch/csrc/autograd/variable.h>"
 C.include "<torch/csrc/autograd/function.h>"
 
+-- inline-c can not parse c++'s symbol like torch::Tensor which includes "::".
+-- "helper.h" defines "typedef torch::Tensor* PtrTensor;" to remove "::".
+C.include "helper.h"
```

Comment:
I think we should improve [this inline-c parser](https://github.com/fpco/inline-c/blob/master/inline-c/src/Language/C/Types/Parse.hs#L238) to identify c++'s symbols.

Reply:
Agreed. It would be a good addition but since `inline-c-cpp` just piggybacks off of `inline-c`. I would imagine that this sort of parser extension might entail a bit of back and forth between the inline-c author(s).

I'm inviting you to the org in case you'd like to work asynchronously against a hasktorch fork of inline-c. You should have repo fork & creation rights with all expected push access. The quick fix would be to change this ADT, directly (as you've pointed out); we can figure out how to correctly push the changes upstream later.

Reply:
Thx!
I will make repo fork, after my cold is gone.